ipam/ring/ring.go

```diff
@@ -576,6 +584,16 @@ func (r *Ring) PeerNames() map[mesh.PeerName]struct{} {
 	return res
 }
 
+func (r *Ring) callUpdateCallback() func() {
```

Comment:
I would put this near the invariant assertion functions.

And perhaps name it `trackUpdates`, seeing as the function itself isn't actually invoking the callback, the returned function is.